drivers/hmis/spec/models/hmis/ce/sql_expression_translator_spec.rb

```
@@ -72,5 +72,45 @@
       result = described_class.call('current_age = ((10 + 5) * (2 + 3))', field_map)
       expect(result.to_sql).to include('((10 + 5) * (2 + 3))')
     end
+
+    context 'with unknown fields' do
```

It looks like "unknown_field" is something that's not resolvable using the field maps, is that right? I would think that we should raise an error in that case? it feels like a potential recipe for silent failures, for example if we have a typo on a cded key name or something, it would always pass? Let me know if I'm misunderstanding this!

---

Sorry this code should be more clear. In the context of the SqlExpressionTranslator, an "unknown field" is one that can't be resolved into SQL, which is the case for all CDE fields currently. Because they aren't filtered in the database, the SQL filter treats them as true.

A typo in a CDE key (e.g., 'cde.custom_assessment.some_key_with_a_typo') will pass the initial SQL filter but should then be caught during the second evaluation phase. The CdeFieldMap should raise if the CDE definition can't be found.

I've adjusted the language to replace 'unknown' with 'unresolved' which hopefully clarifies the intent.

---

Ah I see, thanks for the explainer!